crates/extensions/c8y_mapper_ext/src/operations/firmware_update.rs

```diff
@@ -104,22 +103,41 @@ impl CumulocityConverter {
                     .with_retain()
                     .with_qos(QoS::AtLeastOnce);
 
-                let metadata_topic = self.mqtt_schema.topic_for(
+                let cmd_metadata_topic = self.mqtt_schema.topic_for(
                     topic_id,
                     &Channel::CommandMetadata {
                         operation: OperationType::FirmwareUpdate,
                     },
                 );
-                let metadata_payload = FirmwareMetadata {
+
+                let update_cmd_metadata = Message::new(&cmd_metadata_topic, "{}")
+                    .with_retain()
+                    .with_qos(QoS::AtLeastOnce);
```

Comment:
On successful, command metadata message should not be published. The twin message is enough. So, remove these lines.